presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```diff
@@ -142,4 +133,116 @@ public int size()
             }
         };
     }
+
+    private class KeySet
```

Comment:
But the code that "can go wrong" is already there. At the minimum, all broken methods should throw.

- keySet
- entrySet
- values
- containsValue